src/test/java/org/myrobotlab/service/ServoTest.java

```diff
@@ -273,10 +276,43 @@ public void moveToBlockingTest() throws Exception {
     // log.info("Move to blocking took {} milliseconds", delta);
     assertTrue("Servo should be enabled", servo01.isEnabled());
     assertFalse("Servo should not be moving now.", servo01.isMoving());
-    // Now let's wait for the idle disable timer to kick off + 1000ms
-    // TODO: figure out why smaller values like 100ms cause this test to fail.
-    // there seems to be some lag
-    Thread.sleep(servo01.getIdleTimeout() + 1000);
+    
+    CountDownLatch moveLatch = new CountDownLatch(1);
+    CountDownLatch targetLatch = new CountDownLatch(1);
+    CountDownLatch disableLatch = new CountDownLatch(1);
```

Comment:
I don't see this being counted down anywhere, is it supposed to be?

Reply:
```java

    new Thread(() -> {
        log.info("starting at {}", System.currentTimeMillis());
        servo01.moveTo(0);
        moveLatch.countDown();
    }).start();

    // wait for the move to start
    moveLatch.await();

    // wait for the move to complete using waitTargetPos
    new Thread(() -> {
        servo01.waitTargetPos();
        targetLatch.countDown();
    }).start();
```

Reply:
it is